kore/app/repl/Main.hs

```diff
@@ -203,49 +208,60 @@ mainWithOptions
+                definition <- loadDefinitions [definitionFileName, specFile]
+                indexedModule <- loadModule mainModuleName definition
+                specDefIndexedModule <- loadModule specModule definition
 
-        when (replMode == RunScript && isNothing (unReplScript replScript)) $
-            lift $ do
-                putStrLn
-                    "You must supply the path to the repl script\
-                    \ in order to run the repl in run-script mode."
-                exitFailure
+                let
+                    smtConfig =
+                        SMT.defaultConfig
+                            { SMT.timeOut = smtTimeOut
+                            , SMT.preludeFile = smtPrelude
+                            }
 
-        when (replMode == Interactive && scriptModeOutput == EnableOutput) $
-            lift $ do
-                putStrLn
-                    "The --save-run-output flag is only available\
-                    \ when running the repl in run-script mode."
-                exitFailure
+                when
+                    (  replMode == RunScript
+                    && isNothing (unReplScript replScript)
+                    )
+                    $ lift $ do
+                        putStrLn
+                            "You must supply the path to the repl script\
+                            \ in order to run the repl in run-script mode."
+                        exitFailure
 
-        SMT.runSMT smtConfig $ do
-            give (MetadataTools.build indexedModule)
-                $ declareSMTLemmas indexedModule
-            proveWithRepl
-                indexedModule
-                specDefIndexedModule
-                Nothing
-                mvarLogAction
-                replScript
-                replMode
-                scriptModeOutput
-                outputFile
-                mainModuleName
+                when
+                    (  replMode == Interactive
+                    && scriptModeOutput == EnableOutput
+                    )
+                    $ lift $ do
+                        putStrLn
+                            "The --save-run-output flag is only available\
+                            \ when running the repl in run-script mode."
+                        exitFailure
 
+                SMT.runSMT smtConfig $ do
+                    give (MetadataTools.build indexedModule)
+                        $ declareSMTLemmas indexedModule
+                    proveWithRepl
+                        indexedModule
+                        specDefIndexedModule
+                        Nothing
+                        mvarLogAction
+                        replScript
+                        replMode
+                        scriptModeOutput
+                        outputFile
+                        mainModuleName
+            files <- listDirectory tempDirectory
+            createTarGz (tempDirectory <> ".tar.gz") tempDirectory files
```

Comment:
Let's extract `archiveDirectoryReport` from `kore-exec` so that we can use it here, too.